libmach/test.c

```diff
@@ -4,13 +4,18 @@
 
 typedef void resize_callback(void*, uint32_t, uint32_t);
 
+typedef enum {
+  MachFailure,
+  MachSuccess
+} MachReturn;
```

Comment:
I wonder if it might be a good idea to have these prefixed with an underscore, use a `_Force32` to force the enum to be 32-bits (in case someone puts it somewhere expecting a fixed size), similar to how [webgpu.h does it](https://github.com/webgpu-native/webgpu-headers/blob/64db862b70cee7a5a6dc89dc7d920a0300125f4a/webgpu.h#L150-L158). We could also name this "Status" instead of "Return" so that there could be other error status codes in the future:

```c
typedef enum MachStatus {
    MachStatus_Success = 0x00000000,
    MachStatus_Error = 0x00000001,
    MachStatus_Force32 = 0x7FFFFFFF
} MachStatus;
```

Maybe you have a different idea for error handling though?

Reply:
I'm not exactly an expert on error handling in C... just having error codes is probably fine for now.